src/core/account/account-api.ts

```diff
@@ -498,6 +510,197 @@ export function makeAccountApi(ai: ApiInput, accountId: string): EdgeAccount {
+      walletInfo.keys = { ...publicKeys, ...walletInfo.keys }
+
+      const log = makeLog(ai.props.logBackend, `${walletId}-${walletType}`)
+      let balanceMap: EdgeBalanceMap = new Map()
+      let detectedTokenIds: string[] = []
+      let syncRatio: number = 0
+
+      let needsUpdate = false
+      const updateWallet = (): void => {
+        if (needsUpdate) {
+          update(out)
+          needsUpdate = false
+        }
+      }
+      const updater = makePeriodicTask(async () => {
+        await snooze(1000) // one second
+        updateWallet()
+      }, 0)
+
+      const plugin =
+        ai.props.state.plugins.currency[config.currencyInfo.pluginId]
+      const engine = await plugin.makeCurrencyEngine(walletInfo, {
+        callbacks: {
+          onAddressChanged: () => {},
+          onAddressesChecked: (progressRatio: number) => {
+            if (out.syncRatio === 1) return
+
+            if (progressRatio === 1) {
+              syncRatio = progressRatio
+              needsUpdate = true
+            }
+          },
+          onNewTokens: (tokenIds: string[]) => {
+            const sortedTokenIds = tokenIds.sort((a, b) => a.localeCompare(b))
+
+            if (detectedTokenIds.length !== sortedTokenIds.length) {
+              detectedTokenIds = sortedTokenIds
+              needsUpdate = true
+              return
+            }
+            for (let i = 0; i < sortedTokenIds.length; i++) {
+              if (detectedTokenIds[i] !== sortedTokenIds[i]) {
+                detectedTokenIds = sortedTokenIds
+                needsUpdate = true
+                return
+              }
+            }
+          },
+          onStakingStatusChanged: () => {},
+          onTokenBalanceChanged: (tokenId: EdgeTokenId, balance: string) => {
+            if (balanceMap.get(tokenId) === balance) return
+
+            balanceMap = new Map(balanceMap)
+            balanceMap.set(tokenId, balance)
+            needsUpdate = true
+          },
+          onTransactionsChanged: () => {},
+          onTxidsChanged: () => {},
+          onUnactivatedTokenIdsChanged: () => {},
+          onWcNewContractCall: () => {},
+          onBlockHeightChanged: () => {},
+          onBalanceChanged: () => {}
+        },
+        customTokens: { ...config.customTokens },
+        enabledTokenIds: [...Object.keys(config.allTokens)],
+        log,
+        userSettings: { ...(config.userSettings ?? {}) },
+        walletLocalDisklet: makeMemoryDisklet(),
+        walletLocalEncryptedDisklet: makeMemoryDisklet()
+      })
+
+      const {
+        unsafeBroadcastTx = false,
+        unsafeMakeSpend = false,
+        unsafeSyncNetwork = false
+      } = plugin.currencyInfo
+
+      const privateKeys = { ...keys }
+
+      const syncNetwork = async (): Promise<void> => {
+        try {
+          syncNetworkTask.start({ wait: false })
+          await syncNetworkTask
+        } catch (error) {
+          syncNetwork().finally(() => {})
+        }
+      }
```

Comment:
Get rid of this whole block, and just do `syncNetworkTask?.start({ wait: false })` directly in the `startEngine` method. There is no need to `await syncNetworkTask` because it's not a promise. Also, the start & stop methods can't throw - errors come out through the `onError` callback.